### ISSUE TITLE
Q(lambda) target computation drops the value needed for the second-to-last step, corrupting the whole trajectory's targets

### ISSUE THREAD

**Issue by epignatelli** [mttga/purejaxql#12]
## What's wrong

In `_get_target` (e.g. `purejaxql/pqn_gymnax.py`, and the equivalent block in the other `pqn_*.py` scripts), the backward-scan Q(λ) target computation drops the one Q-value it needs for the second-to-last target, and silently substitutes the wrong one in its place - which then contaminates every earlier target through the λ-mixing term.

```python
last_q = network.apply(..., transitions.next_obs[-1], train=False)
last_q = jnp.max(last_q, axis=-1)
...
last_q = last_q * (1 - transitions.done[-1])
lambda_returns = transitions.reward[-1] + config["GAMMA"] * last_q
_, targets = jax.lax.scan(
    _get_target,
    (lambda_returns, last_q),          # <- seeds next_q with Q(s_T)
    jax.tree_util.tree_map(lambda x: x[:-1], transitions),   # <- drops transitions[T-1]
    reverse=True,
)
```

`last_q = Q(s_T)` (the bootstrap evaluated after the *whole* rollout) is correctly the one-step bootstrap for `targets[T-1]`. But it's *also* seeded as the `next_q` the scan carries into its first call, which computes `targets[T-2]` - and the correct one-step bootstrap for `targets[T-2]` is `Q(s_{T-1})`, i.e. `max(transitions.q_val[T-1], axis=-1)`. That value is never used: `transitions.q_val[T-1]` is exactly the entry `[:-1]` slices away. `Q(s_T)` gets substituted for it instead, and since `targets[T-2]` then feeds into `targets[T-3]` via `delta = lambda_returns - next_q`, the error propagates backward through the entire trajectory - only `targets[T-1]` comes out correct.

## Minimal reproduction

Self-contained (only needs `jax` + `rlax`, `pip install jax rlax`), no repo checkout required - `_get_target`/the surrounding target computation below is copied verbatim from `pqn_gymnax.py`, only de-classed into a free function:

```python
from typing import NamedTuple
import jax, jax.numpy as jnp, numpy as np, rlax

class Transition(NamedTuple):
    reward: jnp.ndarray
    done: jnp.ndarray
    q_val: jnp.ndarray

def calculate_targets(transitions, last_q, gamma, lam):
    def _get_target(lambda_returns_and_next_q, transition):
        lambda_returns, next_q = lambda_returns_and_next_q
        target_bootstrap = transition.reward + gamma * (1 - transition.done) * next_q
        delta = lambda_returns - next_q
        lambda_returns = target_bootstrap + gamma * lam * delta
        lambda_returns = (1 - transition.done) * lambda_returns + transition.done * transition.reward
        next_q = jnp.max(transition.q_val, axis=-1)
        return (lambda_returns, next_q), lambda_returns

    last_q = last_q * (1 - transitions.done[-1])
    lambda_returns = transitions.reward[-1] + gamma * last_q
    _, targets = jax.lax.scan(
        _get_target, (lambda_returns, last_q),
        Transition(transitions.reward[:-1], transitions.done[:-1], transitions.q_val[:-1]),
        reverse=True,
    )
    return jnp.concatenate((targets, lambda_returns[np.newaxis]))

# a fixed "true" Q(s_i) for a 6-step, single-env, no-termination trajectory
rng = np.random.default_rng(1)
T = 6
Q = rng.normal(size=(T + 1,)).astype(np.float32) * 3       # Q(s_0) .. Q(s_T)
reward = rng.normal(size=(T,)).astype(np.float32)
done = np.zeros(T, dtype=np.float32)
gamma, lam = 0.9, 0.8

q_val = jnp.asarray(Q[:-1])[:, None, None]                 # q_val[i] = Q(s_i), shape (T, num_envs=1, num_actions=1)
transitions = Transition(jnp.asarray(reward)[:, None], jnp.asarray(done)[:, None], q_val)
targets = np.asarray(calculate_targets(transitions, jnp.asarray([Q[T]]), gamma, lam)).ravel()

# ground truth: rlax.lambda_returns, DeepMind's independently-implemented
# Q(lambda) (its docstring documents this exact setup: "Q(lambda): v_t = max(q_t, axis=-1)")
next_values = jnp.asarray(Q[1:])                             # V(s_{t+1}) for every t
discount = gamma * (1.0 - jnp.asarray(done))
ground_truth = np.asarray(rlax.lambda_returns(jnp.asarray(reward), discount, next_values, lam))

print("pqn_gymnax targets:", targets)
print("rlax ground truth: ", ground_truth)
print("abs diff:           ", np.abs(targets - ground_truth))
```

Output:

```
pqn_gymnax targets: [ 1.0425059   0.02460287 -0.72000766 -0.43116036 -1.3173257  -2.1862278 ]
rlax ground truth:  [ 1.1852053   0.22279646 -0.4447388  -0.04884233 -0.78632873 -2.1862278 ]
abs diff:            [0.14269948 0.19819358 0.27526885 0.38231802 0.530997   0.        ]
```

Only the last (boundary) target matches; every earlier one is off, growing further from the trajectory's end as the seeding error propagates backward.

## Suggested fix

Don't slice off `transitions[-1]` before the scan - keep the full-length arrays as `xs`, and seed the carry's `next_q` slot with something that's only ever used for a genuine "one step past the rollout" bootstrap, not reused as the second-to-last target's one-step value. Concretely, the scan needs `q_val[T-1]` available as an input to compute `targets[T-2]` correctly, and today it never is.

Thanks for the paper and the clean codebase - happy to be told I'm missing something about the intended semantics here, but the numerical divergence against `rlax.lambda_returns` (whose docstring documents this exact target as "Peng's Q(lambda)") seems hard to explain otherwise.
